### PR TITLE
Update AWS AMP integration to support multi-cluster launch

### PR DESCRIPTION
**Pull request by linhlam-kc**
Rewrite the AWS AMP integration to support multi-cluster launch on 06/15. The setup was tested and verified by AWS team. cc: @AjayTripathy @alexkubecost @robfaraj

### CODE REVIEW

File: aws-amp-integration.md [hunk 1]
```diff
@@ -9,136 +9,271 @@ Kubecost leverages the open-source Prometheus project as a time series database
 * [Amazon Managed Prometheus (AMP)](https://docs.aws.amazon.com/prometheus/latest/userguide/what-is-Amazon-Managed-Service-Prometheus.html)
 * [AMP IAM permissions and policies](https://docs.aws.amazon.com/prometheus/latest/userguide/AMP-and-IAM.html)
 
-## Install AMP
+## Architecture
+The architecture of this integration is similar to Amazon EKS cost monitoring with Kubecost, which is described in the previous blog post, with some enhancements as follows:
 
-### Prerequisites
+In this integration, an additional AWS SigV4 container is added to the cost-analyzer pod, acting as a proxy to help query metrics from Amazon Managed Service for Prometheus using the AWS SigV4 signing process. It enables passwordless authentication to reduce the risk of exposing your AWS credentials.
```
Comment (thomasvn) [inline aws-amp-integration.md:15]: Is there any reason why you're no longer abbreviating Amazon Managed Prometheus to AMP in this doc?

File: aws-amp-integration.md [hunk 2]
```diff
@@ -9,136 +9,271 @@ Kubecost leverages the open-source Prometheus project as a time series database
...
-    --cluster ${YOUR_CLUSTER_NAME} --region ${AWS_REGION} \
-    --approve
+![Quick-Start architecture](https://raw.githubusercontent.com/kubecost/docs/main/images/aws_amp_multi_small.png)
+
+### Federated architecture
+To support the large-scale infrastructure, Kubecost uses Amazon S3 services to improve the query performance efficiently. On top of the Amazon Prometheus Workspace, Kubecost stores the Kubecost's extract, transform, and load (ETL) data in a central S3 bucket. Kubecost's ETL data is a computed cache based on Prometheus's metrics, from which the customers can perform all possible Kubecost queries. By storing the ETL data on an S3 bucket, this integration offers resiliency to your cost allocation data, improves the performance and enables high availability architecture for your Kubecost setup. 
+
+The following architecture diagram illustrates the large-scale infrastructure setup:
+
+![Federated architecture](https://raw.githubusercontent.com/kubecost/docs/main/images/aws_amp_multi_large.png)
```
Comment (thomasvn) [inline aws-amp-integration.md:33]:
```suggestion
![Federated architecture](./images/aws_amp_multi_large.png)
```

File: aws-amp-integration.md [hunk 3]
```diff
@@ -9,136 +9,271 @@ Kubecost leverages the open-source Prometheus project as a time series database
...
+### Quick-Start architecture
+The small-scale infrastructure can manage a small multi-cluster setup. The following architecture diagram illustrates the small-scale infrastructure setup:
 
-eksctl utils associate-iam-oidc-provider \
-    --cluster ${YOUR_CLUSTER_NAME} --region ${AWS_REGION} \
-    --approve
+![Quick-Start architecture](https://raw.githubusercontent.com/kubecost/docs/main/images/aws_amp_multi_small.png)
+
+### Federated architecture
+To support the large-scale infrastructure, Kubecost uses Amazon S3 services to improve the query performance efficiently. On top of the Amazon Prometheus Workspace, Kubecost stores the Kubecost's extract, transform, and load (ETL) data in a central S3 bucket. Kubecost's ETL data is a computed cache based on Prometheus's metrics, from which the customers can perform all possible Kubecost queries. By storing the ETL data on an S3 bucket, this integration offers resiliency to your cost allocation data, improves the performance and enables high availability architecture for your Kubecost setup.
```
Comment (thomasvn) [inline aws-amp-integration.md:29]:
```suggestion
To support a large-scale environment, Kubecost leverages a [Federated ETL architecture](./federated-etl.md). In addition to Amazon Prometheus Workspace, Kubecost stores its extract, transform, and load (ETL) data in a central S3 bucket. Kubecost's ETL data is a computed cache based on Prometheus's metrics, from which users can perform all possible Kubecost queries. By storing the ETL data on an S3 bucket, this integration offers resiliency to your cost allocation data, improves the performance and enables high availability architecture for your Kubecost setup. 
```

Thoughts on this change?

File: aws-amp-integration.md [hunk 4]
```diff
@@ -9,136 +9,271 @@ Kubecost leverages the open-source Prometheus project as a time series database
...
-* You have IAM credentials to create AMP and IAM roles programmatically.
-* You have an existing Amazon EKS cluster with OIDC enabled. You can consider using the following command to enable OIDC for your existing Amazon EKS cluster:
+When the Amazon Managed Service for Prometheus integration is enabled, the bundled Prometheus server in the Kubecost Helm Chart is configured in the remote_write mode. The bundled Prometheus server sends the collected metrics to Amazon Managed Service for Prometheus using the AWS SigV4 signing process. All metrics and data are stored in Amazon Managed Service for Prometheus, and Kubecost queries the metrics directly from Amazon Managed Service for Prometheus instead of the bundled Prometheus. It helps customers not worry about maintaining and scaling the local Prometheus instance.
 
-{% hint style="info" %}
-Remember to replace `<YOUR_CLUSTER_NAME>` and `<AWS_REGION>` with your desired values.
-{% endhint %}
+There are two architectures you can deploy: 
+The Quick-Start architecture supports a small multi-cluster setup.
+The Federated architecture supports a large multi-cluster setup.
```
Comment (bstuder99) [inline aws-amp-integration.md:21]: Would recommend we format this as bullets or a numbered list of options

File: aws-amp-integration.md [hunk 5]
```diff
@@ -9,136 +9,271 @@ Kubecost leverages the open-source Prometheus project as a time series database
...
-        "statusCode": "CREATING"
-    },
-    "tags": {},
-    "workspaceId": "${AMP_WORKSPACE_ID}"
-}
+```bash
+export AMP_WORKSPACE_ID=$(aws amp list-workspaces --region ${AWS_REGION} --output json --query 'workspaces[?alias==`kubecost-amp`].workspaceId | [0]' | cut -d'"' -f 2)
+echo $AMP_WORKSPACE_ID
 ```
+### Set up the environment:
```
Comment (bstuder99) [inline aws-amp-integration.md:67]: "Setting up the environment"

File: aws-amp-integration.md [hunk 6]
```diff
@@ -9,136 +9,271 @@ Kubecost leverages the open-source Prometheus project as a time series database
...
+#### Step 1: set environment variables for integrating Kubecost with Amazon Managed Service for Prometheus
 
-The workspace should be created in a few seconds. You can log in to the [AWS AMP console](https://console.aws.amazon.com/prometheus/) to retrieve more information. You need to set `$REMOTEWRITEURL` and `$QUERYURL` for use in the integration with Kubecost later as follows:
+Run the following command to set environment variables for integrating Kubecost with Amazon Managed Service for Prometheus
```
Comment (bstuder99) [inline aws-amp-integration.md:70]: Needs a colon at the end of this sentence

File: aws-amp-integration.md [hunk 7]
```diff
@@ -9,136 +9,271 @@ Kubecost leverages the open-source Prometheus project as a time series database
...
+export RELEASE="kubecost"
+export YOUR_CLUSTER_NAME=<YOUR_EKS_CLUSTER_NAME>
+export AWS_REGION=${AWS_REGION}
+export VERSION="1.103.4"
+export KC_BUCKET="kubecost-etl-metrics" # Remove this line if you want to set up small-scale infrastructure 
+export AWS_ACCOUNT_ID=$(aws sts get-caller-identity --query Account --output text)
+export REMOTEWRITEURL="https://aps-workspaces.${AWS_REGION}.amazonaws.com/workspaces/${AMP_WORKSPACE_ID}/api/v1/remote_write"
+export QUERYURL="http://localhost:8005/workspaces/${AMP_WORKSPACE_ID}"
+```
+#### Step 2: set up S3 bucket, IAM policy and Kubernetes secret for storing Kubecost ETL files
```
Comment (bstuder99) [inline aws-amp-integration.md:82]: "Step 2: Set..."

File: aws-amp-integration.md [hunk 8]
```diff
@@ -9,136 +9,271 @@ Kubecost leverages the open-source Prometheus project as a time series database
...
 
-## Install Kubecost with the default values
+Note: You can ignore this step 2 for the small-scale infrastructure setup
```
Comment (bstuder99) [inline aws-amp-integration.md:84]: "> **Note**: You can ignore Step 2 when setting up for small-scale infrastructure."

**Comment by bstuder99** [comment]
Agree with Thomas' suggestions and provided several of my own

File: aws-amp-integration.md (hunk 1 above)
Comment (linhlam-kc) [reply aws-amp-integration.md:15]: Per AWS marketing requirement, I keep it that way in all documentation I wrote for AWS integration to not fix it later.

File: aws-amp-integration.md (hunk 4 above)
Comment (linhlam-kc) [reply aws-amp-integration.md]: Thank you Brett ! Fixed in the latest commit.

File: aws-amp-integration.md [hunk 9]
```diff
@@ -9,136 +9,271 @@ Kubecost leverages the open-source Prometheus project as a time series database
...
-    },
-    "tags": {},
-    "workspaceId": "${AMP_WORKSPACE_ID}"
-}
+```bash
+export AMP_WORKSPACE_ID=$(aws amp list-workspaces --region ${AWS_REGION} --output json --query 'workspaces[?alias==`kubecost-amp`].workspaceId | [0]' | cut -d'"' -f 2)
+echo $AMP_WORKSPACE_ID
 ```
+### Set up the environment:
+#### Step 1: set environment variables for integrating Kubecost with Amazon Managed Service for Prometheus
```
[the remark this answers is not in the document]
Comment (linhlam-kc) [reply aws-amp-integration.md]: Thank you Brett ! Fixed in the latest commit.

File: aws-amp-integration.md (hunk 6 above)
Comment (linhlam-kc) [reply aws-amp-integration.md]: Thank you Brett ! Fixed in the latest commit.

File: aws-amp-integration.md (hunk 7 above)
Comment (linhlam-kc) [reply aws-amp-integration.md]: Thank you Brett ! Fixed in the latest commit.

File: aws-amp-integration.md (hunk 8 above)
Comment (linhlam-kc) [reply aws-amp-integration.md]: Thank you Brett ! Fixed in the latest commit.

**Review by bstuder99 — APPROVED** [review]

**Pull request merged**